sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -180,23 +181,21 @@ def __call__(
         """This call is expected to accept the same inputs than sklearn's private
         _kmeans_single_lloyd and produce the same outputs.
         """
-        # Input validation
-        X, centers_init, dtype = self._set_dtype(X, centers_init)
-
-        work_group_size = (
-            self.work_group_size_multiplier * self.preferred_work_group_size_multiple
-        )
-
-        n_features = X.shape[1]
-        n_samples = X.shape[0]
-        n_clusters = centers_init.shape[0]
+        (
+            X,
+            cluster_centers,
+            dtype,
+            work_group_size,
+            n_features,
+            n_samples,
+            n_clusters,
+        ) = self._check_inputs(X, centers_init)
 
         # Create a set of kernels
         (
             n_centroids_private_copies,
-            fixed_window_fused_kernel,
-            fixed_window_assignment_kernel,
-        ) = make_kmeans_fixed_window_kernels(
+            fused_lloyd_fixed_window_single_step,
+        ) = make_lloyd_single_step_fixed_window_kernel(
```

Comment:
Has the _kernel suffix be removed intentionally here? Also, there is a discrepancy w.r.t the name of the factory. Conventionally, I would have expected those two names:

```suggestion
            fused_lloyd_fixed_window_single_step_kernel,
        ) = make_fused_lloyd_fixed_window_single_step_kernel(
```

Does this makes sense?

Reply:
I thought that the readability of the loop in this function (the "high level code") is better with names that only describe what the function does functionally (thinking of inputs/outputs) rather than other characteristics (like its nature of being a kernel)

Reply:
This makes sense to me. Do you think it's possible to have a name which covers both aspect (i.e. what this performs and it being a kernel)?

Reply:
no :shrug: but if you think it's better i'll add _kernel prefix everywhere

Reply:
Either is fine with me.